src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.Client/ProjectProperties/ProjectPropertiesEditorFactory.cs

```diff
@@ -0,0 +1,168 @@
+            out IntPtr ppunkDocView,
+            out IntPtr ppunkDocData,
+            out string? pbstrEditorCaption,
+            out Guid pguidCmdUI,
+            out int pgrfCDW)
+        {
+            // TODO mouse wait cursor throughout this operation
+            // TODO try/catch all this (dispose designerLoader on catch, if not null)
+
+#pragma warning disable RS0030 // Do not used banned APIs
+            ThreadHelper.ThrowIfNotOnUIThread();
+#pragma warning restore RS0030 // Do not used banned APIs
+
+            ppunkDocView = IntPtr.Zero;
+            ppunkDocData = IntPtr.Zero;
+
+            // An empty caption allows the project name to be used as the caption.
+            pbstrEditorCaption = "";
+
+            pguidCmdUI = default;
+            pgrfCDW = default;
+
+            if ((grfCreateDoc & (VSConstants.CEF_OPENFILE | VSConstants.CEF_SILENT)) == 0)
+            {
+                Debug.Fail("Must be opening the file (not cloning it) and creating the editor silently");
+                return HResult.InvalidArg;
+            }
+
+            _paneData = new ProjectPropertiesWindowPaneData(pvHier, ServiceProvider.GlobalProvider);
+
+            ppunkDocData = Marshal.GetIUnknownForObject(_paneData);
+
+            var newEditor = new ProjectPropertiesWindowPane(pvHier);
+            ppunkDocView = Marshal.GetIUnknownForObject(newEditor);
+
+            const _VSRDTFLAGS windowFlags = _VSRDTFLAGS.RDT_DontAddToMRU |    // TODO review this
+                                            _VSRDTFLAGS.RDT_CantSave |        // TODO review this -- we do want to save
+                                            _VSRDTFLAGS.RDT_ProjSlnDocument |
+                                            _VSRDTFLAGS.RDT_VirtualDocument |
+                                            _VSRDTFLAGS.RDT_DontAutoOpen;     // TODO review this
+
+            pgrfCDW = (int)windowFlags;
+
+            pbstrEditorCaption = string.Empty; // no need to add anything to the caption
+                
+            return HResult.OK;
+        }
+
+        public int MapLogicalView(ref Guid rguidLogicalView, out string? pbstrPhysicalView)
+        {
+            // The default physical view for an editor must be null, and we only support one physical view.
+            pbstrPhysicalView = null;
+
+            // Disallow TextView, as that suggests the requested view is the XML editor.
+            if (rguidLogicalView == VSConstants.LOGVIEWID.TextView_guid)
+            {
+                return HResult.NotImplemented;
+            }
+
+            if (rguidLogicalView == VSConstants.LOGVIEWID.Primary_guid)
+            {
+                return HResult.OK;
+            }
+
+            if (rguidLogicalView == VSConstants.LOGVIEWID.Designer_guid)
+            {
+                return HResult.OK;
+            }
+
+            // We may also be called with other GUIDs for specific pages in the UI. Rather than list them all here,
```

Comment:
NOTE: This is actually used for Debug -> [Project] Debug Properties command.

Reply:
Thanks. When invoked that way we see GUID [`0273c280-1882-4ed0-9308-52914672e3aa`](https://github.com/dotnet/project-system/search?q=0273c280-1882-4ed0-9308-52914672e3aa&unscoped_q=0273c280-1882-4ed0-9308-52914672e3aa). I'll capture this info in the code.

We are considering pulling the debug launch settings out into its own UI in future. We'll have to handle this code path at that point I expect.

We will verify whether menu item/command is marked as Codespaces-compliant.